storage-resize-images/functions/src/index.ts

```diff
@@ -61,142 +63,88 @@ const generateResizedImageHandler = async (
-
+  let failed = null;
   let localOriginalFile: string;
   let remoteOriginalFile: File;
-  try {
-    localOriginalFile = path.join(os.tmpdir(), uuidv4());
-    const tempLocalDir = path.dirname(localOriginalFile);
-
-    // Create the temp directory where the storage file will be downloaded.
-    !verbose || logs.tempDirectoryCreating(tempLocalDir);
-    await mkdirp(tempLocalDir);
-    !verbose || logs.tempDirectoryCreated(tempLocalDir);
-
-    // Download file from bucket.
-    remoteOriginalFile = bucket.file(filePath);
-    !verbose || logs.imageDownloading(filePath);
-    await remoteOriginalFile.download({ destination: localOriginalFile });
-    !verbose || logs.imageDownloaded(filePath, localOriginalFile);
 
-    // Get a unique list of image types
-    const imageTypes = new Set(config.imageTypes);
+  try {
+    // Download the original file
+    [localOriginalFile, remoteOriginalFile] = await downloadOriginalFile(
+      bucket,
+      filePath,
+      verbose
+    );
 
-    // Convert to a set to remove any duplicate sizes
-    const imageSizes = new Set(config.imageSizes);
+    // Check content filter and replace with placeholder if needed
+    const filterResult = await processContentFilter(
+      localOriginalFile,
+      object,
+      bucket,
+      verbose,
+      config
+    );
 
-    const tasks: Promise<ResizedImageResult>[] = [];
+    // Process image resizing if content filter didn't fail
+    if (filterResult.failed !== true) {
+      const resizeResults = await resizeImages(
+        bucket,
+        localOriginalFile,
+        parsedPath,
+        objectMetadata
+      );
 
-    imageTypes.forEach((format) => {
-      imageSizes.forEach((size) => {
-        tasks.push(
-          modifyImage({
-            bucket,
-            originalFile: localOriginalFile,
-            parsedPath,
-            contentType: object.contentType,
-            size,
-            objectMetadata: objectMetadata,
-            format,
-          })
-        );
+      await events.recordSuccessEvent({
+        subject: filePath,
+        data: {
+          input: object,
+          outputs: resizeResults,
+          contentFilterPassed: filterResult.passed,
+        },
       });
-    });
-
-    const results = await Promise.allSettled(tasks);
-
-    await events.recordSuccessEvent({
-      subject: filePath,
-      data: { input: object, outputs: results },
-    });
 
-    const failed = results.some(
-      (result) => result.status === "rejected" || result.value.success === false
-    );
+      // Only update failed status if it's still null (not already failed from content filter)
+      failed =
+        filterResult.failed === null
+          ? resizeResults.some(
+              (result) =>
+                result.status === "rejected" || result.value.success === false
+            )
+          : filterResult.failed;
+    } else {
+      failed = true;
+    }
 
+    // Handle failure or success
     if (failed) {
       logs.failed();
```

Comment:
we should include the reason for failure (either content filter failed to work, or resize failed)